ompi/communicator/communicator.h

```diff
@@ -233,7 +234,7 @@ typedef struct ompi_communicator_t ompi_communicator_t;
  * This is done so we don't end up needing a structure that is
  * incredibly larger than necessary because of the bitness.
  */
-#define PREDEFINED_COMMUNICATOR_PAD (sizeof(void*) * 192)
+#define PREDEFINED_COMMUNICATOR_PAD (sizeof(void*) * 194)
```

Comment:
In discussing this more with @gpaulsen I'm now seeking clarification myself. Maybe @bosilca and @jsquyres can clarify how (if at all) the `PREDEFINED_COMMUNICATOR_PAD` should be changed when the communicator changes.

For example:
```
Assume:
  sizeof(void*) = 8 bytes
  ompi_communicator_t = 80 bytes
  PREDEFINED_COMMUNICATOR_PAD = 800 bytes (sizeof(void*) * 100)
Then:
  ompi_predefined_communicator_t = 80 + (800 - 80) = 800 bytes
```

So if I were to add 1 pointer to the `ompi_communicator_t` and **not** change `PREDEFINED_COMMUNICATOR_PAD` that works out to:
```
Assume:
  sizeof(void*) = 8 bytes
  ompi_communicator_t = 88 bytes
  PREDEFINED_COMMUNICATOR_PAD = 800 bytes (sizeof(void*) * 100)
Then:
  ompi_predefined_communicator_t = 88 + (800 - 88) = 800 bytes
```

So from my understanding I don't think the padding should ever have to change until the `sizeof(ompi_communicator_t)` becomes larger than `PREDEFINED_COMMUNICATOR_PAD`.

Is that logic correct or are we missing something here?